pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -4236,96 +4237,117 @@ func TestChangefeedEnrichedSourceWithData(t *testing.T) {
-				return func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
-					clusterID := s.Server.ExecutorConfig().(sql.ExecutorConfig).NodeInfo.LogicalClusterID().String()
-
-					sqlDB := sqlutils.MakeSQLRunner(s.DB)
-
-					sqlDB.Exec(t, `CREATE TABLE foo (i INT PRIMARY KEY)`)
-					sqlDB.Exec(t, `INSERT INTO foo values (0)`)
-					testFeed := feed(t, f, fmt.Sprintf(`CREATE CHANGEFEED FOR foo WITH envelope=enriched, enriched_properties='source', format=%s`, testCase.format))
-					defer closeFeed(t, testFeed)
-
-					var jobID int64
-					var nodeName string
-					var sourceAssertion func(actualSource map[string]any)
-					if ef, ok := testFeed.(cdctest.EnterpriseTestFeed); ok {
-						jobID = int64(ef.JobID())
-					}
-					sqlDB.QueryRow(t, `SELECT value FROM crdb_internal.node_runtime_info where component = 'DB' and field = 'Host'`).Scan(&nodeName)
-
-					sourceAssertion = func(actualSource map[string]any) {
-						var nodeID any
-						if testCase.format == "avro" {
-							actualSourceValue := actualSource["source"].(map[string]any)
-							nodeID = actualSourceValue["node_id"].(map[string]any)["string"]
-						} else {
-							nodeID = actualSource["node_id"]
-						}
-						require.NotNil(t, nodeID)
+		for _, withMVCCTs := range []bool{true, false} {
+			t.Run(fmt.Sprintf("%s/mvcc_ts=%v", testCase.name, withMVCCTs), func(t *testing.T) {
+				clusterName := "clusterName123"
+				dbVersion := "v999.0.0"
+				defer build.TestingOverrideVersion(dbVersion)()
+				mkTestFn := func(sink string) func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+					return func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+						clusterID := s.Server.ExecutorConfig().(sql.ExecutorConfig).NodeInfo.LogicalClusterID().String()
 
-						sourceNodeLocality := fmt.Sprintf(`region=%s`, testServerRegion)
+						sqlDB := sqlutils.MakeSQLRunner(s.DB)
 
-						// There are some differences between how we specify sinks here and their actual names.
-						if sink == "sinkless" {
-							sink = sinkTypeSinklessBuffer.String()
+						sqlDB.Exec(t, `CREATE TABLE foo (i INT PRIMARY KEY)`)
+						sqlDB.Exec(t, `INSERT INTO foo values (0)`)
+						stmt := fmt.Sprintf(`CREATE CHANGEFEED FOR foo WITH envelope=enriched, enriched_properties='source', format=%s`, testCase.format)
+						if withMVCCTs {
+							stmt += ", mvcc_timestamp"
+						}
+						testFeed := feed(t, f, stmt)
+						defer closeFeed(t, testFeed)
+
+						var jobID int64
+						var nodeName string
+						var sourceAssertion func(actualSource map[string]any)
+						if ef, ok := testFeed.(cdctest.EnterpriseTestFeed); ok {
+							jobID = int64(ef.JobID())
 						}
+						sqlDB.QueryRow(t, `SELECT value FROM crdb_internal.node_runtime_info where component = 'DB' and field = 'Host'`).Scan(&nodeName)
+
+						sourceAssertion = func(actualSource map[string]any) {
+							var nodeID any
+							if testCase.format == "avro" {
```

Comment:
nit: use a switch statement instead and explicitly label the `"json"` case

Reply:
is that really meaningfully better?

Reply:
I think it makes it slightly more readable heh, you could also add a default case that fails the test due to an unexpected format but that might be overkill

Reply:
ehhhh i think this is fine how it is ;) ;)